pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -106,6 +106,14 @@ func (n *Node) Add(ctx context.Context, pod *v1.Pod) error {
 	return nil
 }
 
+// FinalizeScheduling is called once all scheduling has completed and allows the node to perform any cleanup
+// necessary before its requirements are used for instance launching
+func (n *Node) FinalizeScheduling() {
```

Comment:
We already have a concept that models this:
```
	// RestrictedLabels are labels that should not be used
	// because they may interfer the internal provisioning logic.
	RestrictedLabels = sets.NewString(
		EmptinessTimestampAnnotationKey,
		v1.LabelHostname,
	)
```